monsieurkayode/More-Recipes-Kayode-Adeola
# #152933201 Implement First Feedback Server Side

#### What does this PR do?
Refactors server-side logic implementation for resource endpoints
#### Description of Task to be completed?
##### Have pagination implemented for the following routes
* `GET /api/v1/recipes`
* `GET /api/v1/users/recipes/favorites`
* `GET /api/v1/recipes/:recipeId/reviews`

##### Make the following implementations of the route `POST /api/v1/users/recipes`
* change how recipe categorization is being implemented when creating a new post
* a user cannot create multiple posts with the same title

##### Check and return response to validation errors for all input field when a user tries to signup
via `POST /api/v1/users`

##### Implement CRUD for reviews resource
* `POST /api/v1/recipes/:recipeId/reviews`
* `PATCH /api/v1/recipes/:recipeId/reviews/:id`
* `GET /api/v1/recipes/:recipeId/reviews/:id`
* `DELETE /api/v1/recipes/:recipeId/reviews/:id`

#### How should this be manually tested?
* Clone the repo and install all dependencies using `npm install` 
* Setup your database, cd into the `config` folder in server and rename to `database` under development in `config.json` to your database name
* Run `npm migrate` to migrate tables
* Start the server  by running `npm start` 
* Add a `.env` file, see readme for sample

> Test the routes above with postman
* Obtain a token from the endpoint `POST /api/v1/signup`
* Set authorization in header using _key_: *x-access-token*, _value_: <TOKEN>
* Populate database with recipe records via `POST /api/v1/recipes`
* Test all routes listed in Tasks description above
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#152933201
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/26303683/33144190-76d8c4b2-cfbc-11e7-9c9d-993ec6ba0a38.png)
![image](https://user-images.githubusercontent.com/26303683/33144238-9e9a08bc-cfbc-11e7-9f49-fd1a84edc382.png)
